### PR TITLE
Add machine learning algorithm based on rusty-machine

### PR DESCRIPTION
**Pull request by z1queue**
1.implement examples of dbscan.
2.implement examples of gaussian_mixture_model.
3.implement examples of gaussian_processes.
4.implement examples of gen_linear_model.
5.implement examples of lin_reg.
6.implement examples of logistic_reg.
7.implement examples of naive_bayes
8.implement examples of neural_net.
9.implement examples of svm.

10.modify private_join_and_compute

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -39,3 +39,22 @@ improvements. Details are discussed
 	  dedicated project called
 [MesaPy](https://github.com/mesalock-linux/mesapy). In this specific example,
 we demonstrate how to invoke the MesaPy engine integrated into MesaTEE.
+* [DBSCAN](DBSCAN)
+	- Provides an implementaton of DBSCAN clustering.
+* [Generalized Linear Model](GeneralizedLinearModel)
+	- Contains implemention of generalized linear models using iteratively reweighted least squares.The model will automatically add the intercept term to the input data.
```
Comment (mssun) [inline examples/README.md]: I suggest to break the line by 80 characters. This is more friendly for diff in version control system like Git.

Comment (mssun) [reply examples/README.md]: Same for other lines in the `README.md`.

**Comment by mssun** [comment]
@litongxin1991, can you help to review this PR? Specifically, can you check whether these functions are properly integrated in MesaTEE. Thanks.

File: mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/naive_bayes.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        model
+            .train(&input.input_model_data, &input.target_model_data)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+        // Predict the classes on the input data
+        let predict = model
+            .predict(&input.input_model_data)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+
+        let mut output = String::new();
+        writeln!(&mut output, "{:?}", predict)
```
Comment (litongxin1991) [inline mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/naive_bayes.rs]: The output would be ```Matrix { rows: 6, cols: 3, data: [1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0] }```
Can you format the result, not directly return the debug output?

File: examples/neural_net/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    target_data format:
+        1.
+        0.
+        1.
+        0.
+        0.
+        1.
+        ....
+    test_data format:
+        f32,f32,f32,f32 ...
```
Comment (litongxin1991) [inline examples/neural_net/src/main.rs]: The test data can contains multiple lines. Please change the test_data format in the usage.

File: mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/neural_net.rs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            .train(&input.input_model_data, &input.target_model_data)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+        // predict a new test data
+        let predict = model
+            .predict(&input.test_data)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+
+        let mut output = String::new();
+        writeln!(&mut output, "{:?}", predict)
+            .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
```
Comment (litongxin1991) [inline mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/neural_net.rs]: Please use "OutputGenerationError". Other workers have same problems. Please fix them as well.

File: mesatee_services/fns/sgx_trusted_lib/src/global.rs [hunk 5]
```diff
@@ -55,7 +57,7 @@ pub fn save_file_for_file_owner(
 }
 
 pub fn register_trusted_worker_statically() {
-    for _i in 0..10 {
+    for _i in 0..23 {
```
Comment (litongxin1991) [inline mesatee_services/fns/sgx_trusted_lib/src/global.rs]: The number is the max concurrent running tasks at the same time for each worker.  It doesn't need to be changed when you add new workers.

**Comment by litongxin1991** [comment]
Some examples don't verify the result. Although the result may be random numbers, shall we use other ways to check the result, like the format of the result?

**Comment by z1queue** [comment]
> Some examples don't verify the result. Although the result may be random numbers, shall we use other ways to check the result, like the format of the result?

i verfify the result of example/gaussian_mixture_model use the shell command sort. 
and the other example neural_net output result is not very easy to verfify,the number of the output all is random, so i only just output it.

**Pull request merged**